aten/src/ATen/native/vulkan/ops/Upsample.cpp

```diff
@@ -0,0 +1,80 @@
+#include <ATen/native/UpSample.h>
+#include <ATen/native/vulkan/ops/Common.h>
+#include <torch/library.h>
+
+namespace at {
+namespace native {
+namespace vulkan {
+namespace ops {
+namespace {
+
+Tensor upsample_nearest2d(
+    const Tensor& input_arg,
+    const IntArrayRef output_sizes,
+    const c10::optional<double> scales_h,
+    const c10::optional<double> scales_w) {
+  api::Context* const context = api::context();
+
+  const Tensor input = input_arg.is_vulkan() ? input_arg : input_arg.vulkan();
+  const vTensor& v_input = convert(input);
+
+  vTensor v_output{
+    context,
+    output_sizes,
```

Comment:
Since `output_sizes` only has 2 elements, I also think this should be something like

`{input_arg.sizes()[0], input_arg.sizes()[1], output_sizes[0], output_sizes[1]`